app/src/androidTest/java/com/example/jasmiensofiecels/wordly/view/dailyWord/ValidDailyWordActivityTest.java

```diff
@@ -0,0 +1,189 @@
+                .check(matches(atPositionOnView(0, withText("(in science fiction) a robot with a human appearance"), R.id.word_definition)));
+    }
+
+    @Test
+    //The purpose of the this test is to ensure that the first recyclerView list order number is correct
+    public void firstWordOrder_onRecyclerView() {
+        //Then
+        onView(withId(R.id.recycler_view))
+                .perform(RecyclerViewActions.scrollToPosition(0))
+                .check(matches(atPositionOnView(0, withText("1"), R.id.word_list_order)));
+
+    }
+
+    @Test
+    //The purpose of the this test is to ensure that the second recyclerView list definition is correct
+    public void secondWordDefinition_onRecyclerView() {
+        //Then
+        onView(withId(R.id.recycler_view))
+                .perform(RecyclerViewActions.scrollToPosition(1))
+                .check(matches(atPositionOnView(1, withText("an open-source operating system used for smartphones and tablet computers"), R.id.word_definition)));
+    }
+
+    @Test
+    //The purpose of the this test is to ensure that the second recyclerView list order number is correct
+    public void secondWordOrder_onRecyclerView() {
+        //Then
+        onView(withId(R.id.recycler_view))
+                .perform(RecyclerViewActions.scrollToPosition(1))
+                .check(matches(atPositionOnView(1, withText("2"), R.id.word_list_order)));
+    }
+
+
+    //Methods to aid test assertions
+    //Thank you Stack Over Flow!
+    public class RecyclerViewItemCountAssertion implements ViewAssertion {
+        private final int expectedCount;
+
+        public RecyclerViewItemCountAssertion(int expectedCount) {
+            this.expectedCount = expectedCount;
+        }
+
+        @Override
+        public void check(View view, NoMatchingViewException noViewFoundException) {
+            if (noViewFoundException != null) {
+                throw noViewFoundException;
+            }
+
+            RecyclerView recyclerView = (RecyclerView) view;
+            assertThat(recyclerView.getAdapter().getItemCount(), is(expectedCount));
+        }
+    }
+
+
+    public static Matcher<View> atPositionOnView(final int position, final Matcher<View> itemMatcher,
+                                                 @NonNull final int targetViewId) {
+
+        return new BoundedMatcher<View, RecyclerView>(RecyclerView.class) {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("has view id " + itemMatcher + " at position " + position);
+            }
+
+            @Override
+            public boolean matchesSafely(final RecyclerView recyclerView) {
+                RecyclerView.ViewHolder viewHolder = recyclerView.findViewHolderForAdapterPosition(position);
+                View targetView = viewHolder.itemView.findViewById(targetViewId);
+                return itemMatcher.matches(targetView);
+            }
+        };
+    }
+
```

Comment:
I really like your separation of positive scenarios and negative scenarios in different classes.

Reply:
I also love how much documentation you put on the tests. You seem to have covered all scenarios

Reply:
I'm thinking back to how you've separated the positive and negatives into separate classes. I really like this. However, you're actually only testing one positive scenario but have 8 different tests. Each test should be self-contained. So having `android` still in the text box between tests shouldn't happen. You should be testing all of these in a single test, which would be `valid string` or `android entered`. 

I always look at the example Android docs if I'm unsure about something. See how each test is a different scenario:
https://github.com/googlesamples/android-architecture/blob/dev-todo-mvvm-live/todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/AppNavigationTest.java

They have also tested if pressing back exits the app from both screens. Interesting, I don't usually test for that.

Reply:
This is probably a more relevant example as it involves user input:

https://github.com/googlesamples/android-architecture/blob/dev-todo-mvvm-live/todoapp/app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksScreenTest.java